### PR TITLE
171648120 Page authenticates a user with Google and Facebook

### PR DESCRIPTION
**Pull request by Irutingabo**
#### Description
The feature provides a page where a user sign in using Google or Facebook
 
#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [X] Unit
- [ ] Integration
- [ ] End-to-end
- [ ] No testing required

#### How to Test

Give instructions on how reviewers can verify your work

#### Any background context you want to add (Optional)

Screenshots or GIFs

#### STORY ID

[Finishes #171648120]

### CODE REVIEW

File: src/Components/Auth/SocialAuth/SocialAuth.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      border: none;
+      height: 53px;
+      margin: 3px;
+      outline: none;
+      width: 53px;
+    }
+
+    &__logos.google {
+      background-image: url("../../../Assets/googleLogo.svg");
+    }
```
Comment (Agnes-N) [inline src/Components/Auth/SocialAuth/SocialAuth.scss]: Fix these hound issues!

**Review by Agnes-N — CHANGES_REQUESTED** [review]

**Review by Agnes-N — DISMISSED** [review]

File: .eslintrc.json [hunk 2]
```diff
@@ -19,6 +19,11 @@
     "linebreak-style": 0,
     "global-require": 0,
     "eslint linebreak-style": [0, "error", "windows"],
+    "jsx-a11y/no-noninteractive-element-interactions": [0],
+    "jsx-a11y/click-events-have-key-events": [0],
+    "jsx-a11y/control-has-associated-label": [0],
+    "no-case-declarations": 0,
+    "react/button-has-type": [0],
```
Comment (placideirandora) [inline .eslintrc.json]: Turn them back on and refer to the documentation.

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    expect(socialAuth.find("button").length).toEqual(2);
+    expect(socialAuth.find("span").length).toEqual(1);
+    expect(socialAuth.find("div").length).toEqual(2);
+  });
+
+  it("should test google button click event", () => {
+    window.open = jest.fn();
+    const socialAuth = mount(<SocialAuth />);
+    socialAuth.find("button.social-signin-section__google").simulate("click");
+    expect(window.open).toBeCalled();
```
Comment (placideirandora) [inline __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Use **toHaveBeenCalled()** instead.

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  it("should test facebook button click event", () => {
+    window.open = jest.fn();
+    const socialAuth = mount(<SocialAuth />);
+    socialAuth.find("button.social-signin-section__facebook").simulate("click");
+    expect(window.open).toBeCalled();
```
Comment (placideirandora) [inline __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Use **toHaveBeenCalled()** instead.

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+import SocialAuth from "../../../../src/Components/Auth/SocialAuth/SocialAuth";
+
+describe("<SocialAuth />", () => {
```
Comment (placideirandora) [inline __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Put **Social Auth Tests**

File: __tests__/utils/tokenVerify.test.js [hunk 6]
```diff
@@ -1,20 +1,39 @@
-import tokenVerify from "../../src/Utils/tokenVerify";
+import tokenVerify, { getTokenFromParams } from "../../src/Utils/tokenVerify";
 
 describe(" TokenVerify test suite", () => {
```
Comment (placideirandora) [inline __tests__/utils/tokenVerify.test.js]: Put **Token Verification Tests**

File: __tests__/utils/tokenVerify.test.js [hunk 7]
```diff
@@ -1,20 +1,39 @@
...
-  it("should return false if no or wrong token is passed", () => {
-    const returned = tokenVerify();
-    expect(returned).toBe(false);
+  describe(" Verifies token in Sign In", () => {
```
Comment (placideirandora) [inline __tests__/utils/tokenVerify.test.js]: Put **Local Sign In Token Verification**

File: __tests__/utils/tokenVerify.test.js [hunk 8]
```diff
@@ -1,20 +1,39 @@
...
-    const returned = tokenVerify(token);
-    expect(returned).toBe(false);
-  });
-  it("should return true when provided token is correct and not expired", () => {
-    const token =
-      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NywiZW1haWwiOiJicmlhbi5pbmV6YUBnbWFpbC5jb20iLCJpYXQiOjE1ODY1MTQ0ODh9.hQyEoAzddqte4Tr6nhM5ufkXA7-PoNXw59YV8QVjHvs";
-    const returned = tokenVerify(token);
-    expect(returned).toBe(true);
+
+  describe("Verifies token in Social authentication", () => {
```
Comment (placideirandora) [inline __tests__/utils/tokenVerify.test.js]: Put **Social Auth Token Verification**

File: package.json [hunk 9]
```diff
@@ -43,7 +42,6 @@
     "eslint-plugin-prettier": "^3.1.3",
     "eslint-plugin-react": "^7.20.0",
     "eslint-plugin-react-hooks": "^2.5.0",
-    "file-loader": "^6.0.0",
     "html-webpack-plugin": "^4.3.0",
     "jest": "^26.0.1",
     "jest-styled-components": "^7.0.2",
```
Comment (placideirandora) [inline package.json]: Remove this package.

**Review by placideirandora — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 2 above)
Comment (Irutingabo) [reply .eslintrc.json]: Feedback implemented

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js (hunk 3 above)
Comment (Irutingabo) [reply __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Implemented

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js (hunk 4 above)
Comment (Irutingabo) [reply __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Implemented

File: __tests__/Components/Auth/SocialAuth/SocialAuth.test.js (hunk 5 above)
Comment (Irutingabo) [reply __tests__/Components/Auth/SocialAuth/SocialAuth.test.js]: Feedback implemented

File: __tests__/utils/tokenVerify.test.js (hunk 6 above)
Comment (Irutingabo) [reply __tests__/utils/tokenVerify.test.js]: Feedback Implemented

File: __tests__/utils/tokenVerify.test.js (hunk 7 above)
Comment (Irutingabo) [reply __tests__/utils/tokenVerify.test.js]: Feedback implemented

File: __tests__/utils/tokenVerify.test.js (hunk 8 above)
Comment (Irutingabo) [reply __tests__/utils/tokenVerify.test.js]: FeedBack Implemented!

File: package.json (hunk 9 above)
Comment (Irutingabo) [reply package.json]: Done!

**Review by placideirandora — APPROVED** [review]

**Pull request merged**